src/middlewared/middlewared/plugins/system/debug.py

```diff
@@ -62,16 +63,21 @@ def progress_callback(percent, desc):
         except Exception as e:
             raise CallError(f'Failed to generate debug: {e!r}')
 
-    @accepts()
+    @accepts(
+        Dict(
+            'debug_options',
+            Dict('extra', additional_attrs=True)
```

Comment:
Don't like this option because of how it's implemented. This allows arbitrary data (unbounded in size) to be written to an arbitrary path inside the debug archive (so it could overwrite existing files).

1. We should add a top-level directory to `ixdiagnose` (i.e. another Plugin) that gets added by default (maybe like `WebUIDir` or something). It should capture all data in there by default. (glob pattern)
2. The top-level directory name should be statically defined in middleware repo or in ixdiagnose repo and then imported accordingly
3. The api args that we accept should be a file name, and then a Pipe object
4. We need to truncate the contents of the file to be written to 50KB (Per request of webUI)

Reply:
> so it could overwrite existing files

It can't (ixdiagnose won't do it)

> It should capture all data in there by default. (glob pattern)

I don't understand what does this mean in this context, what's "all data"?

Reply:
Because the webUI team is not the only consumer of this endpoint. If we still want to add this feature then we need to restrict the starting component of the path to `plugins/`.

Reply:
@yocalebo I don't understand what part of my message are you replying to. So, to conclude, what needs to be done now?